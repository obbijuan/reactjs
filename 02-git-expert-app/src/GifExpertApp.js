import React, { useState } from 'react'
import { CategoryAdd } from './components/CategoryAdd'
import { GifGrid } from './components/GifGrid'

export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [Categories, setCategories] = useState(defaultCategories)

    return (
        <>
            <h2>GifExpertApp</h2>
            <CategoryAdd setCategories={ setCategories }/>
            <hr/>

            <ol>
                {
                    Categories.map(category => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    )
                    )
                }
            </ol>
        </>
    )
}
