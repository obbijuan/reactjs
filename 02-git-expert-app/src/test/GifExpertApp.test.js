import React from 'react';
import { GifExpertApp } from '../GifExpertApp';
import { shallow } from 'enzyme';

describe('Pruebas para el componente <GifExpertApp/>', () => {

    test('debe mostrarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe mostrar un lista de categorias', () => {
        const categories = ['Matrix', 'Star Wars'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
    
    
    
})
