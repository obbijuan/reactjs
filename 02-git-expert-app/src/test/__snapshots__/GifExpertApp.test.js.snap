// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente <GifExpertApp/> debe mostrar un lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Matrix"
      key="Matrix"
    />
    <GifGrid
      category="Star Wars"
      key="Star Wars"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas para el componente <GifExpertApp/> debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
